x-pack/legacy/plugins/apm/server/lib/transactions/get_transaction_by_trace/index.ts

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import {
+  PROCESSOR_EVENT,
+  TRACE_ID,
+  PARENT_ID,
+  SERVICE_NAME,
+  TRACE,
+  TRANSACTION,
+  TIMESTAMP
+} from '../../../../common/elasticsearch_fieldnames';
+import { Transaction } from '../../../../typings/es_schemas/ui/Transaction';
+import { Setup } from '../../helpers/setup_request';
+import { ProcessorEvent } from '../../../../common/processor_event';
+
+export async function getTransactionByTraceId(traceId: string, setup: Setup) {
+  const { client, indices } = setup;
+  const params = {
+    index: indices['apm_oss.transactionIndices'],
+    body: {
+      _source: [TIMESTAMP, TRANSACTION, SERVICE_NAME, TRACE],
```

Comment:
I don't think it's useful to use `_source` if we specify most of the properties manually. Question is: do we need everything? Afaict we just need `trace.id, transaction.id, transaction.name, transaction.type`.

Reply:
I also need the `service.name`. I thought `_source` returned only what is specified in the array, isn't it?

Reply:
As talked, I'm going to return the whole Transaction.